nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -226,8 +226,10 @@ BtNavigator::navigateToPose()
       topic_logger.flush();
     };
 
+  // Make sure that the Bt is not in a running state from a previous execution
+  bt_->haltAllActions(tree_.root_node);
```

Comment:
I think last beef I have -- this should be after run so that once the BT exits, we reset the state rather than leaving the BT in a dirty state and cleaning up just before a run.

Reply:
I am not 100% sure about this one. In practice, unless a particular node (not the root) has been stopped from the "outside", the tree will always complete in a "good state".

ControlNodes and DecoratorNodes are responsible for this clean-up of their children, therfore this is propagated through the entire tree.

I prefer to keep it like this.

Reply:
well by dirty state I mean between runs, if we inspect the tree, it _may_ still have running nodes. While it *shouldnt* if things are implemented properly, if a user improperly created a node that ignores or doesn't properly handle the return, the halt command at the end of a navigation request would be a second chance for the code to actually stop what its doing in background threads. 

For safety, I would insist on having this afterwards. Not for issues in your BT library, but in issues potentially present in user code. Also, the halt all actions may add latency to a navigation request.